### ISSUE TITLE
DataSet.IsRefreshRequired is always true

### ISSUE THREAD

**Issue by Mylan719** [riganti/dotvvm#598]
I have following code to fill DataSet for my grid. The datased is bound to gridview and a pager controls.
```csharp
  private async Task ReloadDataAsync()
        {
            await ExecuteSafeAsync(async () =>
            {
                await listFacade.FillDataSetAsync(DataSet, Filter);
                DataSet.IsRefreshRequired = false;
            });
        }

        public override Task Init()
        {
            if (!Context.IsPostBack)
            {
                DataSet = new GridViewDataSet<TGridDto>()
                {
                    PagingOptions =
                {
                    PageSize = 10
                },
                    SortingOptions = new SortingOptions
                    {
                        SortExpression = "Name"
                    }
                };
            }
            return base.Init();
        }

        public override async Task PreRender()
        {
            await ExecuteSafeAsync(async () =>
            {
                if (!Context.IsPostBack || DataSet?.IsRefreshRequired == true)
                {
                    await ReloadDataAsync();
                }
                await base.PreRender();
            });
        }
```

However the `IsRefreshRequired` property of `DataSet` is always true. Even if I just loaded the data set it to `false`. If then I make another random request that has nothing to do with the grid `IsRefreshRequired`  is `true` again.

**Comment by Mylan719** [comment]
This is probably a bug under 1.1.9 false is sent as is should as lon as I updated to DotVVM 2.0 it's broken:
Repro here:
[DotvvmApplication5.zip](https://github.com/riganti/dotvvm/files/1996350/DotvvmApplication5.zip)

**Comment by adamjez** [comment]
It seems fixed in the current latest stable version (2.0.3).
